S/SHTns/build_tarballs.jl

```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "SHTns"
+version = v"3.5.1"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://bitbucket.org/nschaeff/shtns/downloads/shtns-$(version).tar.gz", "77f8a33b94df8786d2ce9b95cbfbe548f00443625b310b8c64012b22c8a7394f")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd shtns/
+export CFLAGS="-fPIC"
+ ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j ; make install
+gcc -shared -o libshtns.so *.o -lfftw3
+cp libshtns.so $prefix/lib/
+install_license LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "freebsd"; )
+]
```

Comment:
```suggestion
platforms = supported_platforms()
```

Reply:
Hello @giordano, thanks for taking the time to review this recipe.
Will this test all platforms ? The build fails on some platforms, the list was produced by the wizard after trying all platforms except Mac. If you prefer to see it by yourself, no problem, I'll do the commit.

Reply:
Yes, let's see what happens on all platforms :slightly_smiling_face: